pkg/ext-proc/backend/provider.go

```diff
@@ -81,11 +81,21 @@ func (p *Provider) Init(refreshPodsInterval, refreshMetricsInterval time.Duratio
 		for {
 			time.Sleep(refreshMetricsInterval)
 			if err := p.refreshMetricsOnce(); err != nil {
-				klog.V(1).Infof("Failed to refresh metrics: %v", err)
+				klog.V(4).Infof("Failed to refresh metrics: %v", err)
 			}
 		}
 	}()
 
+	// Periodically print out the pods and metrics for DEBUGGING.
+	if klog.V(2).Enabled() {
+		go func() {
+			for {
+				time.Sleep(5 * time.Second)
+				klog.Infof("===DEBUG: Current Pods and metrics: %+v", p.AllPodMetrics())
```

Comment:
Why not print this as part of the refresh loop above instead of spawning a separate go routine?

Reply:
The refresh loop is run much more frequently (20ms in my benchmark).